#8729 Map views plugin dev documentation (#8774)
geosolutions-it/MapStore2: docs/developer-guide/maps-configuration.md

- `offsetTime` the end of a time range
- `timelineData` timeline options
- `selectedLayer` selected layer id; if not present time cursor will be unlocked
- `mapViews` map views options

> ### mapViews

Reviewer: I should move the whole section `## Additional map configuration options` at the end of the page.
The priority is anyway on the layers.

Author: @offtherailz done

          "title": "Clip",
          "id": "layer.id.03"
        }
      }
    ]
  }
}
```

The mapViews properties

| Name | Type | Description |
| --- | --- | --- |
| active | boolean | if true the view will be active at initialization |
| selectedId | string | id of the selected view |
| views | array | array of views configuration (see below) |
| resources | array | resources array of resources configuration (see below) |

View configuration object

| Name | Type | Description |
| --- | --- | --- |
| id | string | identifier of the view |
| title | string | title of the view |
| description | string | an html string to describe the view |
| duration | number | time duration during navigation in seconds |
| flyTo | boolean | enable animation transition during navigation |
| center | object | center target position as { latitude (degrees), longitude (degrees), height (meter) } |
| cameraPosition | object | cameraPosition point of view position as { latitude (degrees), longitude (degrees), height (meter) } |
| zoom | number | zoom level |
| bbox | array | bounding box in WGS84 as [minx, miny, maxx, maxy] |
| mask | object | optional configuration for the 3D tiles mask |

> | mask.enabled | boolean | enable mask |

Reviewer: ```suggestion
| mask.enabled | boolean | enable mask |
```
```suggestion
| mask.enabled | boolean | if true enables the mask |
```

> | mask.resourceId | string | identifier of a resource configuration |

Reviewer: ```suggestion
| mask.resourceId | string | identifier of a resource configuration in the `resources` array |
```

> | mask.inverse | boolean | enabled inverse mask |

Reviewer: ```suggestion
| mask.inverse | boolean | if true enables the inverse mask |
```

> | mask.offset | number | offset in meter for the inverse mask |

Reviewer: ```suggestion
| mask.offset | number | offset in meters for the inverse mask |
```

> | terrain | object | optional configuration for terrain clipping |

Reviewer: ```suggestion
| terrain | object | optional configurations for terrain clipping |
```

> |terrain.clippingLayerResourceId | string | identifier of a resource configuration |

Reviewer: ```suggestion
|terrain.clippingLayerResourceId | string | identifier of a resource configuration in the `resources` array |
```

> | terrain.clippingPolygonFeatureId | string | identifier of a polygon feature available in the selected layer source |

Reviewer: ```suggestion
| terrain.clippingPolygonFeatureId | string | identifier of a polygonal feature available in the selected layer source to use to apply the clipping |
```

| terrain.clippingPolygonUnion | boolean | if true it applies inverse clipping |
| globeTranslucency | object | optional configuration for the globe translucency |

> | globeTranslucency.enabled | boolean | enable translucency |

Reviewer: ```suggestion
| globeTranslucency.enabled | boolean | if true enables translucency |
```

| globeTranslucency.opacity | number | opacity of the globe translucency, it should be a value between 0 and 1 where 1 is fully opaque  |

> | globeTranslucency.fadeByDistance | boolean | if true the translucency is visible only in between the nearDistance and farDistance values |

Reviewer: ```suggestion
| globeTranslucency.fadeByDistance | boolean | if true the translucency is visible only between the `nearDistance` and `farDistance` values |
```

> | globeTranslucency.nearDistance | number | minimum distance to apply translucency when fadeByDistance is true |

Reviewer: ```suggestion
| globeTranslucency.nearDistance | number | when `fadeByDistance` is true it indicates the minimum distance to apply translucency |
```

> | globeTranslucency.farDistance | number | maximum distance to apply translucency when fadeByDistance is true |

Reviewer: ```suggestion
| globeTranslucency.farDistance | number |  when `fadeByDistance` is true it indicates the maximum distance to apply translucency |
```

> | layers | array | array of layer configuration overrides, default property visibility and opacity |

Reviewer: ```suggestion
| layers | array | array of layer configuration overrides, default properties override `visibility` and `opacity` |
```
